tests/unit/backend/corpora/common/utils/test_s3_utils.py

```diff
@@ -1,8 +1,9 @@
 import unittest
 
+import boto3
 from moto import mock_s3
 
-from backend.corpora.common.utils.s3_utils import generate_file_url
+from backend.corpora.common.utils.s3_utils import generate_file_url, head_file
 
 
 class TestS3Utils(unittest.TestCase):
```

Comment:
Use `CorporaTestCaseUsingMockAWS` which mocks out S3 and other AWS resources. It even has a function to generate S3 objects. That will allow you to delete a ton of this code in this file and DRYs up the codebase.